### PR TITLE
[#922] Search sentinels by polygon drawing

### LINKED ISSUE #922: Search sentinels by polygon drawing
Where to find feature?
1. Login
2. Go to `sentinel search on map`
3. Click on map and move mouse

- Draw polygon (as closed form) to search sentinels
- Start drawing new one by clicking on map

### PR DESCRIPTION
**Pull request by danielkryska**
Where to find feature?

    Login
    Go to sentinel search on map
    Click on map and move mouse

- Draw polygon (as closed form) to search sentinels
- Start drawing new one by clicking on map

Closes #922

### CODE REVIEW

**Comment by kmarszalek** [comment]
right now on pr instance you can draw many polygons and it screwing the search.. [search:sentinel-1]
Zrzut ekranu 2020-11-24 o 07 06 04

**Review by kmarszalek — CHANGES_REQUESTED** [review]
web tests and multiple polygons

**Comment by danielkryska** [comment]
Multiple polygons don't influence searching, because the `/dhus/search` endpoint is triggered at the end of drawing actual one.

**Comment by kmarszalek** [comment]
ok, so multiple just doesnt look good or clear.
but, poligon is screwing search. also, you trigger search when you close polygon, why?
search, it doesnt matter if i check sentinel-1 in search or not, if i am drawing polygon i got MSG and not Sentinel. 
only if i take step back and click search button i got sentinel scenes.

**Comment by danielkryska** [comment]
From now polygons should be removed (only one will be visible at time) and footprint is sent on click `search` button.

**Review by jswk — DISMISSED** [review]
The problems mentioned by @kmarszalek seem to be fixed on e1b94b9, however, there's no way I've found to unset the polygon. I'd expect a button to unset the polygon somewhere.

Also, after I search using polygon, navigate to products list and back I'd expect the polygon to have been unset (especially since it's not rendered on map anymore). Currently it is still sent with search requests.

**Review by abacz — APPROVED** [review]

**Comment by Marcelinna** [comment]
after drawing a polygon and clicking search button, search results do not appear
image

File: s4e-web/src/app/views/map-view/sentinel-search/sentinel-search.component.ts [hunk 1]
```diff
@@ -104,7 +104,7 @@ export class SentinelSearchComponent implements OnInit, OnDestroy {
       return;
     }
 
-    this.service.search(this.form).subscribe();
+    this.service.search(this.form);
```
Comment (michal-szostak) [inline s4e-web/src/app/views/map-view/sentinel-search/sentinel-search.component.ts:107]: why remove subscribe?

**Review by michal-szostak — CHANGES_REQUESTED** [review]

**Review by michal-szostak — APPROVED** [review]

**Pull request merged**